test/testers.jl

```diff
@@ -263,4 +265,29 @@ end
         frule_test(iterfun, (x, ẋ))
         rrule_test(iterfun, randn(), (x, x̄))
     end
+
+    @testset "test quaternion non-standard scalar" begin
+        function FiniteDifferences.to_vec(q::Quaternion)
+            function Quaternion_from_vec(q_vec)
+                return Quaternion(q_vec[1], q_vec[2], q_vec[3], q_vec[4])
+            end
+            return [q.s, q.v1, q.v2, q.v3], Quaternion_from_vec
+        end
```

Comment:
We should move this to be defined in the package itself.

Reply:
Do you mean define this in Quaternions or FiniteDifferences?

Reply:
Or ChainRulesTestUtils?

Reply:
FiniteDifferences

Reply:
I don't know if it makes sense to make Quaternions an optional dependency for FiniteDifferences. Since I am only defining this for the purpose of testing, I'm comfortable with being type-piratical but just in the test suite where it can't pollute the methods table for other users. Thoughts?

Reply:
Hey @oxinabox what do you think?